local-setup/jupyter/dataloader/crs_loader.py

```
@@ -555,32 +555,25 @@ def _update_citymodule_db(self, module_code: str, tenant_code: str, current_conf
-              AND tenantid = 'pg'
-              AND data->>'code' = :'mod_code';
-            """
+            # Escape single quotes for SQL string literal
+            new_data = json_lib.dumps(new_config).replace("'", "''")
+            sql = (
+                f"UPDATE eg_mdms_data "
+                f"SET data = '{new_data}'::jsonb, "
+                f"lastmodifiedtime = EXTRACT(EPOCH FROM NOW())::bigint * 1000 "
+                f"WHERE schemacode = 'tenant.citymodule' "
+                f"AND tenantid = 'pg' "
+                f"AND data->>'code' = '{module_code}';"
```

Ruff (0.15.9)

[error] Possible SQL injection vector through string-based query construction

(S608)